Cron/Invalidate.php

```diff
@@ -0,0 +1,168 @@
+ *
+ * @package Nosto\Tagging\Cron
+ */
+class Invalidate
+{
+    /** @var Logger */
+    protected $logger;
+
+    /** @var CacheCollectionFactory */
+    private $cacheCollectionFactory;
+
+    /** @var NostoAccountHelper */
+    private $nostoAccountHelper;
+
+    /** @var CacheRepository */
+    private $cacheRepository;
+
+    /** @var TimezoneInterface */
+    private $timezoneInterface;
+
+    /** @var int */
+    private $intervalHours;
+
+    /** @var int */
+    private $productLimit;
+
+    /**
+     * Invalidate constructor.
+     *
+     * @param Logger $logger
+     * @param CacheCollectionFactory $cacheCollectionFactory
+     * @param CacheRepository $cacheRepository
+     * @param NostoAccountHelper $nostoAccountHelper
+     * @param TimezoneInterface $timezoneInterface
+     * @param int $intervalHours
+     * @param int $productLimit
+     */
+    public function __construct(
+        Logger $logger,
+        CacheCollectionFactory $cacheCollectionFactory,
+        CacheRepository $cacheRepository,
+        NostoAccountHelper $nostoAccountHelper,
+        TimezoneInterface $timezoneInterface,
+        $intervalHours,
+        $productLimit
+    ) {
+        $this->logger = $logger;
+        $this->cacheCollectionFactory = $cacheCollectionFactory;
+        $this->cacheRepository = $cacheRepository;
+        $this->nostoAccountHelper = $nostoAccountHelper;
+        $this->timezoneInterface = $timezoneInterface;
+        $this->intervalHours = $intervalHours;
+        $this->productLimit = $productLimit;
+    }
+
+    /**
+     * Executes cron job
+     * @throws Exception
+     */
+    public function execute()
+    {
+        if (!$this->productLimit || !is_numeric($this->productLimit)) {
+            $this->logger->debug(
+                sprintf(
+                    'Product limit for invalidate cron is set to %s. Not invalidating any products.',
+                    $this->productLimit
+                )
+            );
+            return;
+        }
+        $stores = $this->nostoAccountHelper->getStoresWithNosto();
+        foreach ($stores as $store) {
+            $productCacheCollection = $this->getCollection($store);
+            $updatedCount = $this->cacheRepository->markAsIsDirtyItemsByStore($productCacheCollection, $store);
+            $this->logger->debug(
+                sprintf(
+                    'Invalidated (set dirty) %d products for store %s by invalidate cron.' .
+                            ' Product limit is %d and interval is %d',
+                    $updatedCount,
+                    $store->getCode(),
+                    $this->productLimit,
+                    $this->intervalHours
+                )
+            );
+
+        }
+    }
+
+    /**
+     * @param Store $store
+     * @return CacheCollection
+     * @throws Exception
+     */
+    private function getCollection(Store $store)
+    {
+        return $this->cacheCollectionFactory->create()
```

Comment:
I need to understand why this is inlined and not in the repo.

Reply:
Moved to repo